lib/Runtime/ByteCode/ByteCodeEmitter.cpp

```diff
@@ -2694,7 +2694,12 @@ void ByteCodeGenerator::EmitFunctionBody(FuncInfo *funcInfo)
                         Output::Print(_u("--- DelayCapture: Committed symbol '%s' to slot.\n"), sym->GetName());
                         Output::Flush();
                     }
+                    // REVIEW[ianhall]: HACK to work around this causing an error due to sym not yet being initialized
+                    // what is this doing? Why are we assigning sym to itself?
+                    bool old = sym->GetNeedDeclaration();
+                    sym->SetNeedDeclaration(false);
```

Comment:
Thanks for calling this out, @dilijev.

@pleath I meant to ask you in person about this on Friday but then I was sick.  Do you have any ideas about how to handle this better, or is this a reasonable solution?